core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -492,6 +492,18 @@ object ZIOSpec extends ZIOBaseSpec {
         } yield assertCompletes
       }
     ),
+    suite("fromFutureInterrupt")(
+      testM("running Future can be interrupted") {
+        for {
+          p1     <- ZIO.effectTotal(scala.concurrent.Promise[Unit])
+          p2     <- ZIO.effectTotal(scala.concurrent.Promise[Unit])
+          fiber  <- ZIO.fromFutureInterrupt(ec => p1.future.map(p2.success)(ec)).fork
```

Comment:
I would create a future chain that is infinite, e.g.:

```scala
def infiniteFuture(implicit ec: ExecutionContext): Future[Nothing] = 
  Future(()).flatMap(_ => infiniteFuture)
```

This way you know it will never terminate unless called with `fromFutureInterrupt`.

Reply:
I thought about doing that. How would you verify that it did terminate? If we installed a finalizer on the effect wouldn't it be triggered on interruption even if the future didn't terminate.

Reply:
The test won't time out if it terminates?

Reply:
Ideally `interrupt` would not return until `Future` as killed, but to verify that:

```scala
def infiniteFuture(ref: AtomicInteger)(implicit ec: ExecutionContext): Future[Nothing] = 
  Future(ref.getAndIncrement()).flatMap(_ => infiniteFuture(ref))
```

Then after interrupt returns, you verify that `ref` doesn't change for some small amount of real time (10 millis or so).

Reply:
All set now.